packages/clerk-js/src/ui/components/OrganizationProfile/MembersSearchForm.tsx

```diff
@@ -1,89 +1,84 @@
 
 type MembersSearchProps = {
   /**
    * Controlled query param state by parent component
    */
   query: GetMembersParams['query'];
-  /**
-   * Controlled input field value by parent component
-   */
-  value: string;
   /**
    * Paginated organization memberships
    */
   memberships: ReturnType<typeof useOrganization>['memberships'];
   /**
-   * Handler for change event on input field
+   * Handler for value changes
    */
-  onSearchChange: (value: string) => void;
+  onChange: (query: string) => void;
   /**
-   * Handler for `query` value changes
+   * Minimum search length to trigger query
    */
-  onQueryTrigger: (query: string) => void;
+  minLength: number;
 };
 
-const membersSearchDebounceMs = 500;
-
-export const MembersSearch = ({ query, value, memberships, onSearchChange, onQueryTrigger }: MembersSearchProps) => {
+export const MembersSearchForm = ({ memberships, onChange, minLength }: MembersSearchProps) => {
   const { t } = useLocalizations();
-
-  const debounceTimer = useRef<ReturnType<typeof setTimeout> | null>(null);
+  const searchField = useFormControl('search', '', {
+    type: 'search',
+    label: '',
+    placeholder: localizationKeys('organizationProfile.membersPage.action__search'),
+  });
 
   const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    const eventValue = event.target.value;
-    onSearchChange(eventValue);
+    const value = event.target.value.trim();
 
-    const shouldClearQuery = eventValue === '';
-    if (shouldClearQuery) {
-      onQueryTrigger(eventValue);
-    }
-  };
+    searchField.onChange(event);
+    onChange(value);
 
-  // Debounce the input value changes until the user stops typing
-  // to trigger the `query` param setter
-  function handleKeyUp() {
-    if (debounceTimer.current) {
-      clearTimeout(debounceTimer.current);
+    if (value.length >= minLength) {
+      searchField.clearFeedback();
+      return;
     }
 
-    debounceTimer.current = setTimeout(() => {
-      onQueryTrigger(value.trim());
-    }, membersSearchDebounceMs);
-  }
+    searchField.setInfo(t(localizationKeys('organizationProfile.membersPage.action__search_minLength')));
+  };
 
   // If search is not performed on a initial page, resets pagination offset
   // based on the response count
   useEffect(() => {
-    if (!query || !memberships?.data) {
+    if (!searchField.value || !memberships?.data) {
       return;
     }
 
     const hasOnePageLeft = (memberships?.count ?? 0) <= ACTIVE_MEMBERS_PAGE_SIZE;
     if (hasOnePageLeft) {
       memberships?.fetchPage?.(1);
     }
-  }, [query, memberships]);
+  }, [searchField.value, memberships]);
 
-  const isFetchingNewData = value && !!memberships?.isLoading && !!memberships.data?.length;
+  const isFetchingNewData = searchField.value && !!memberships?.isLoading && !!memberships.data?.length;
 
   return (
-    <Animated asChild>
-      <Flex sx={{ width: '100%' }}>
+    <Flex
```

Comment:
We should follow up with descriptors for these elements. There currently is no way to target MemberSearchForm elements with the appearance prop.

Reply:
Added a descriptor for the container on https://github.com/clerk/javascript/pull/5159/commits/8e0b49d66077c3ac21ee08e449552c6a1ae66af7

Now we have the following:

![CleanShot 2025-02-14 at 13 58 49](https://github.com/user-attachments/assets/12d72a2e-5b8b-4e8b-9765-3b81e59aa628)